snapcraft/tests/pluginhandler/test_scriptlets.py

```diff
@@ -24,6 +24,16 @@
 
 class ScriptletTestCase(tests.TestCase):
 
+    def test_run_prepare_scriptlet(self):
+        handler = mocks.loadplugin(
+            'test-part', part_properties={'prepare': 'touch before-build'})
+
+        handler.build()
+
+        before_build_file_path = os.path.join(handler.code.build_basedir,
+                                              'before-build')
+        self.assertThat(before_build_file_path, FileExists())
```

Comment:
But this is not testing the order, right?
It would be cool to have a test, maybe integration, that makes sure that prepare runs first, then build, and then install.